library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -681,11 +682,15 @@ public void onWSEvent(Event event) {
     /**
      * the opposite of {@link #disconnectWebSocket()}
      */
-    public void reconnectWebSocket() {
-        if (getUser() == null) {
+    public synchronized void reconnectWebSocket() {
+        User user = getUser() != null ? getUser() : lastUser;
```

Comment:
Why it's added here?
Because it's breaks existing logic:
1. developer calls `disconnect`
2. `getUser` returns null
3. `reconnectWebSocket` is not working anymore

With `lastUser` it maybe reconnected again. `reconnectWebSocket` is public and it can be called outside also.